### PR TITLE
(chore): Update go dependencies

### PR DESCRIPTION
**Pull request by ArthurSens**
## Description

Updating our go dependencies as per our [Release](https://github.com/prometheus-operator/prometheus-operator/blob/main/RELEASE.md#update-go-dependencies) process.

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [X] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note

```

### CODE REVIEW

**Comment by ArthurSens** [comment]
After failing CI, I'm also adding an extra step to our release docs :)

**Comment by simonpasquier** [comment]
still failing?

**Comment by ArthurSens** [comment]
> still failing?

Yeah, couldn't find time to fix it yet. Hopefully in a few hours 😬 

If the fix is obvious to someone, suggestions are welcome

**Comment by simonpasquier** [comment]
seems to be https://github.com/kubernetes/kube-openapi/issues/404

**Comment by simonpasquier** [comment]
TL;DR: k8s.io/kube-openapi needs to stay at v0.0.0-20230501164219-8b0f38b5fd1f

**Comment by ArthurSens** [comment]
Thanks for the help! ❤️

**Review by simonpasquier — APPROVED** [review]
